servicetalk-concurrent-api/src/main/java/io/servicetalk/concurrent/api/Completable.java

```diff
@@ -2269,6 +2285,7 @@ public static Completable anyOf(final Iterable<Completable> completables) {
      */
     final void delegateSubscribe(Subscriber subscriber,
                                  CapturedContext capturedContext, AsyncContextProvider contextProvider) {
+        assert contextProvider.context() == capturedContext.captured();
```

Comment:
Open question: should we keep the asserts? They were nice for finding bugs but some people will run apps with asserts enabled. Even then it might be fine, but you never know.